src/codecs/jpeg/encoder.rs

```diff
@@ -648,6 +656,40 @@ impl<W: Write> JpegEncoder<W> {
 
         Ok(())
     }
+
+    fn write_icc_profile_chunks(&mut self) -> io::Result<()> {
+        if self.icc_profile.is_empty() {
+            return Ok(());
+        }
+
+        const MAX_CHUNK_SIZE: usize = 65533 - 14;
+        const MAX_CHUNK_COUNT: usize = 255;
+        const MAX_ICC_PROFILE_SIZE: usize = MAX_CHUNK_SIZE * MAX_CHUNK_COUNT;
+
+        if self.icc_profile.len() > MAX_ICC_PROFILE_SIZE {
+            return Err(io::Error::new(
+                io::ErrorKind::InvalidInput,
+                "ICC profile too large",
+            ));
+        }
+
+        let num_chunks = ((self.icc_profile.len() + MAX_CHUNK_SIZE - 1) / MAX_CHUNK_SIZE) as u8;
```

Comment:
If you create `self.icc_profile.chunks(MAX_CHUNK_SIZE)` iterator first, then `iter.len()` will give you the number of chunks (it's efficient).